controllers/azuremanagedcontrolplane_controller.go

```diff
@@ -95,12 +95,12 @@ func (amcpr *AzureManagedControlPlaneReconciler) SetupWithManager(ctx context.Co
 		return errors.Wrap(err, "error creating controller")
 	}
 
-	// Add a watch on clusterv1.Cluster object for unpause & ready notifications.
+	// Add a watch on clusterv1.Cluster object for pause/unpause & ready notifications.
 	if err = c.Watch(
 		&source.Kind{Type: &clusterv1.Cluster{}},
 		handler.EnqueueRequestsFromMapFunc(amcpr.ClusterToAzureManagedControlPlane),
-		predicates.ClusterUnpausedAndInfrastructureReady(log),
-		predicates.ResourceNotPausedAndHasFilterLabel(log, amcpr.WatchFilterValue),
+		predicates.Any(log, predicates.ClusterCreateInfraReady(log), predicates.ClusterUpdateInfraReady(log), ClusterUpdatePauseChange(log)),
```

Comment:
For `predicates.Any()`, doesn't it return true if any of the predicates inside return true? If so, in this case shouldn't we be only returning true if all three are true?

Reply:
This was adapted from `ClusterUnpausedAndInfrastructureReady` to essentially remove the `ClusterCreateNotPaused` condition and replace `ClusterUpdateUnpaused` with `ClusterUpdatePauseChange`. https://github.com/kubernetes-sigs/cluster-api/blob/54dfd423d80e42f79c33c0149f9719dba3863f11/util/predicates/cluster_predicates.go#L221

These update predicates only look at one particular change of a resource and not just its updated state, so `ClusterUpdateInfraReady` only passes if the old object's infra is not ready and the updated object's is, and that likely won't correspond to the same update when a Cluster is paused/unpaused, so we need `Any` to check if any of those things happened in a particular update and ignore all other ones.